### PR TITLE
TRI-174   Refactor /me endpoint

### PR DESCRIPTION
**Pull request by Juan-Bianchi**
# Pull Request

## Description
Frontend needs to add Role and User to the /api/user/me endpoint to retrieve data for Home Page. It involves some logic refactor in services.

## Ticket Link
[TRI-174](https://linear.app/tricker-sirius/issue/TRI-174/refactor-me-endpoint)

## Steps to Reproduce
1. Run `docker compose up`
2. Open Swagger documentation endpoint [link](http://localhost:8080/api-docs/#/User/get_api_user_me)
3. Feel free to do some requests to test.

### CODE REVIEW

**Review by FedericoMartucci — APPROVED** [review]

File: src/domains/integration/dto/index.ts [hunk 1]
```diff
@@ -30,10 +30,12 @@ export class ProjectDataDTO {
 
 export class UserRole {
```
Comment (luccar31) [inline src/domains/integration/dto/index.ts:34]: Please, document this class.

File: src/domains/userProjectRole/dto/index.ts [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
+  /** Unique identifier of the user's project role. */
   id: string;
+  /** Identifier of the user associated with the project role. */
   userId: string;
+  /** Identifier of the project associated with the project role. */
   projectId: string;
+  /** Identifier of the role associated with the project role. */
   roleId: string;
+  /** Identifier of the user who emitted the project role. */
   userEmitterId: string;
```
Comment (luccar31) [inline src/domains/userProjectRole/dto/index.ts:17]: I don´t understand this. Why is this necessary? I the pm's id for this case?

File: src/domains/userProjectRole/dto/index.ts [hunk 3]
```diff
@@ -20,10 +38,38 @@ export class UserProjectRoleDTO {
...
+ * Data Transfer Object (DTO) representing input data for creating a user's project role.
+ */
 export interface UserProjectRoleInputDTO {
+  /** Identifier of the user associated with the project role. */
   userId: string;
+  /** Identifier of the project associated with the project role. */
   projectId: string;
+  /** Identifier of the role associated with the project role. */
   roleId: string;
+  /** Identifier of the user who emits the project role. */
```
Comment (luccar31) [inline src/domains/userProjectRole/dto/index.ts:51]: Same question as before.

**Review by luccar31 — CHANGES_REQUESTED** [review]

File: src/domains/userProjectRole/dto/index.ts (hunk 2 above)
Comment (Juan-Bianchi) [reply src/domains/userProjectRole/dto/index.ts:17]: Frontend team asked to include it because in the future they may need it.

File: src/domains/userProjectRole/dto/index.ts (hunk 3 above)
Comment (Juan-Bianchi) [reply src/domains/userProjectRole/dto/index.ts:51]: It is for the same reason

File: src/domains/integration/dto/index.ts (hunk 1 above)
Comment (Juan-Bianchi) [reply src/domains/integration/dto/index.ts:34]: done

**Review by luccar31 — APPROVED** [review]
